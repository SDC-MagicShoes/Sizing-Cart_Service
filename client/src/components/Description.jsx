import React from 'react';
import ReactDom from 'react-dom';
import $ from 'jquery';

class Description extends React.Component {
  constructor(props) {
    super(props);
    this.state = {

    }
  }


  render() {
    console.log(this.props);
    return (
      <div className="description">
        <p>
          <div className="description-paragraph">
            {this.props.description[2]}
          </div>
        </p>
        <div className="description-details">
          <ul class="details">
            Shown: {this.props.description[1]}<br/>
            Style: {this.props.description[0]}
          </ul>
        </div>
      </div> 

    )
  }
}

export default Description;