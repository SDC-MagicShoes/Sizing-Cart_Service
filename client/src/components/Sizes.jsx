import React from 'react';

class Sizes extends React.Component {
  constructor(props) {
    super(props);
  }
  
  
  render() {
    console.log(this.props);
    return (
     <div className="sizes-table">
      {this.props.sizes.map(function(size, i) {
        return <button className='sizes-buttons' key={i}>{size}</button>
      })}
     </div> 
    )
  }
}

export default Sizes;