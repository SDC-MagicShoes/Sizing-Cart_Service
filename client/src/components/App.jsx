import React from 'react';
import axios from 'axios';
import Sizes from './Sizes';
import Description from './Description';

const _ = require('lodash');

class App extends React.Component {
  static sortSizes(a, b) {
    return a - b;
  }

  constructor(props) {
    super(props);

    this.state = {
      sizes: [],
      defaultSizes: _.chunk(
        [7, 7.5, 8, 8.5, 9, 9.5, 10, 10.5, 11, 11.5, 12, 12.5, 13, 14, 15, 16, 17, 18], 5,
      ),
      description: '',
      shoeId: '310805-408',
      curButton: -1,
    };
    this.setCurrent = this.setCurrent.bind(this);
  }

  componentDidMount() {
    this.getSizes();
    this.getDescription();
  }

  getSizes() {
    const { shoeId } = this.state;
    axios.get(`/sizes/${shoeId}`)
      .then((response) => {
        let sizes = response.data;
        sizes = sizes.sort(this.sortSizes);
        this.setState({ sizes });
      });
  }

  getDescription() {
    const { shoeId } = this.state;
    axios.get(`/description/${shoeId}`)
      .then((response) => {
        const description = response.data;
        this.setState({ description });
      });
  }


  setCurrent(idx) {
    const newState = this.state;
    newState.curButton = idx;
    this.setState(newState);
  }

  render() {
    const { sizes, curButton, description, defaultSizes } = this.state;
    return (
      <div id="sizesAndDescription">
        <Sizes sizes={sizes} curButton={curButton} setCurrent={this.setCurrent} defaultSizes={defaultSizes} />
        <div className="cart">
          <button type="button" className="cart-button">Add to Cart</button>
          <button type="button" className="heart-button" title="Add to Wishlist"><i className="material-icons">favorite</i></button>
        </div>
        <Description description={description} />
        <div className="description-details">
          <ul className="details">
            <li>
            Shown:
              { description[1] }
            </li>
            <br />
            <li>
            Style:
              { description[0] }
            </li>
          </ul>
          <div className="description-readmore">Read more</div>
        </div>
      </div>
    );
  }
}

export default App;
