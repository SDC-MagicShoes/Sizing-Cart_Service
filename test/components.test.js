import { configure, shallow, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import App from '../client/src/components/App';
import Sizes from '../client/src/components/Sizes';
import Description from '../client/src/components/Description';

configure({ adapter: new Adapter() });

describe('App', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<App debug />);
    expect(component).toMatchSnapshot();
  });
});

describe('Description', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Description debug />);
    expect(component).toMatchSnapshot();
  });
});

describe('Sizes', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Sizes debug />);
    expect(component).toMatchSnapshot();
  });
});