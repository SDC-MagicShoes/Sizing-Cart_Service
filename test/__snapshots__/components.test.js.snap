// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    debug={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Sizes
          sizes={Array []}
        />,
        <Description
          description=""
        />,
      ],
      "id": "sizesAndDescription",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "sizes": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "description": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Sizes
            sizes={Array []}
          />,
          <Description
            description=""
          />,
        ],
        "id": "sizesAndDescription",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "sizes": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "description": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Description should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Description
    debug={true}
    description={
      Array [
        "no description",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="description-paragraph"
        >
          <p />
        </div>,
        <div
          className="description-details"
        >
          <ul
            className="details"
          >
            Shown:
            <br />
            Style:
            no description
          </ul>
        </div>,
      ],
      "className": "description",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p />,
          "className": "description-paragraph",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul
            className="details"
          >
            Shown:
            <br />
            Style:
            no description
          </ul>,
          "className": "description-details",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Shown:",
              undefined,
              <br />,
              "Style:",
              "no description",
            ],
            "className": "details",
          },
          "ref": null,
          "rendered": Array [
            "Shown:",
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            "Style:",
            "no description",
          ],
          "type": "ul",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="description-paragraph"
          >
            <p />
          </div>,
          <div
            className="description-details"
          >
            <ul
              className="details"
            >
              Shown:
              <br />
              Style:
              no description
            </ul>
          </div>,
        ],
        "className": "description",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p />,
            "className": "description-paragraph",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ul
              className="details"
            >
              Shown:
              <br />
              Style:
              no description
            </ul>,
            "className": "description-details",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Shown:",
                undefined,
                <br />,
                "Style:",
                "no description",
              ],
              "className": "details",
            },
            "ref": null,
            "rendered": Array [
              "Shown:",
              undefined,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              "Style:",
              "no description",
            ],
            "type": "ul",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Sizes should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sizes
    debug={true}
    sizes={
      Array [
        Array [
          "no sizes",
        ],
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="sizes-buttons"
          type="button"
        >
          no sizes
        </button>,
      ],
      "className": "sizes-table",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "size_1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "no sizes",
          ],
          "className": "sizes-buttons",
          "type": "button",
        },
        "ref": null,
        "rendered": Array [
          "no sizes",
        ],
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="sizes-buttons"
            type="button"
          >
            no sizes
          </button>,
        ],
        "className": "sizes-table",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "size_1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "no sizes",
            ],
            "className": "sizes-buttons",
            "type": "button",
          },
          "ref": null,
          "rendered": Array [
            "no sizes",
          ],
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
